framework/test/unit/application.spec.ts

```diff
@@ -150,7 +167,28 @@ describe('Application', () => {
 
 			// Act && Assert
 			expect(() => app.registerPlugin(new MyPlugin())).toThrow(
-				'A plugin with name "my-plugin" already registered.',
+				'A plugin with name "my-plugin" is already registered.',
+			);
+		});
+
+		it('should throw error when module with same name is already registered', () => {
+			// Arrange
+			const { app } = Application.defaultApplication(config);
+			class MyPlugin extends TestPlugin {
+				public get name() {
+					return 'my-plugin';
+				}
+			}
+			class MyModule extends TestModule {
+				public get name() {
+					return 'my-plugin';
+				}
+			}
```

Comment:
These 2 classes are always the same and don't need to be copied in all relevant test cases. You can move them to the top of the file, where `TestModule` and `TestPlugin` are declared.